### PR TITLE
Chore: Automatically infer types for dashgrid connected components 

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
Refactor types for dashgrid components so the types for connected props are automatically inferred. This removes the need to automatically declare types for those props and fixes type errors in strict mode related to connected components.

### CODE REVIEW

File: public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx [hunk 1]
```diff
@@ -44,26 +43,11 @@ interface OwnProps {
...
-  initPanelEditor: typeof initPanelEditor;
-  panelEditorCleanUp: typeof panelEditorCleanUp;
-  setDiscardChanges: typeof setDiscardChanges;
-  updatePanelEditorUIState: typeof updatePanelEditorUIState;
-  updateTimeZoneForSession: typeof updateTimeZoneForSession;
-}
+type PropsFromRedux = ConnectedProps<typeof connector>;
 
-type Props = OwnProps & ConnectedProps & DispatchProps;
+type Props = OwnProps & PropsFromRedux;
```
Comment (torkelo) [inline public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx:50]:
```suggestion
type Props = OwnProps & ConnectedProps<typeof connector>;
```

PropsFromRedux type seems unnecessary as it's only used there

**Review by torkelo — APPROVED** [review]
Nice! this looks great! :heart:

**Review by dprokop — APPROVED** [review]
Great work @Clarity-89! Any chance you could describe this pattern in our [Redux style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/redux.md)?

**Comment by Clarity-89** [comment]
> Great work @Clarity-89! Any chance you could describe this pattern in our [Redux style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/redux.md)?

Sure, will do!

**Pull request merged**